iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -659,19 +651,43 @@ popo::InterfacePortData* PortManager::acquireInterfacePortData(capro::Interfaces
     }
 }
 
+void PortManager::publishCurrentServiceRegistry() const noexcept
+{
+    // Send the new serviceRegistry here and ring the bell to inform all ServiceDiscovery instances about the change
+    if (m_serviceRegistryPublisherPortData)
+    {
+        PublisherPortUserType publisher(m_serviceRegistryPublisherPortData);
+        auto maybeChunkHeader = publisher.tryAllocateChunk(sizeof(ServiceRegistry),
+                                                           alignof(ServiceRegistry),
+                                                           CHUNK_NO_USER_HEADER_SIZE,
+                                                           CHUNK_NO_USER_HEADER_ALIGNMENT);
+
+        if (!maybeChunkHeader.has_error())
+        {
+            auto sample = static_cast<ServiceRegistry*>(maybeChunkHeader.value()->userPayload());
+
+            // Copy the complete registry
+            *sample = m_serviceRegistry;
```

Comment:
We must ensure it is not concurrently updated. This happens in the discovery thread only. As long as we only publish in this thread as we are fine.

Reply:
So far, I think we're clean here. Will double check.

Reply:
This method here is indirectly called via `deletePortsOfProcess()`, which is called by `removeProcessAndDeleteRespectiveSharedMemoryObjects()`, which is called if a process is unresponsive, and `monitorProcesses()`, which is called cyclically every 100ms. As this is the usually process/port management I'm unsure if I oversaw something or if I discovered a bug? :thinking:

Reply:
We are safe then and should avoid mutexes for this (not needed, performance). However, these subtle assumptions are easy to get lost and maybe a comment is in order for anyone changing the code.